qiskit_ibm_runtime/transpiler/passes/basis/fold_rzz_angle.py

```diff
@@ -244,3 +248,163 @@ def _quad4(angle: float, qubits: Tuple[Qubit, ...]) -> DAGCircuit:
             check=False,
         )
         return new_dag
+
+
+def convert_to_rzz_valid_pub(
+    program_id: str, pub: Union[SamplerPubLike, EstimatorPubLike]
```

Comment:
If this is intended to be user-facing, I feel like it would be smoother to allow the function to accept a primitive instance instead of a `program_id`. I don't think that the `program_id` is an externally understood attribute (and it probably shouldn't be), and you can always fetch it from the primitive.

```suggestion
    primitive: Union[SamplerV2, EstimatorV2], pub: Union[SamplerPubLike, EstimatorPubLike]
```

Reply:
Done in e0190510. There is a problem though, that defining a primitive instance requires `mode`. It may confuse users, since in practice we're not using this information.